### ISSUE TITLE
Feature Request - Automatically add new Row after tabbing on the last cell

### ISSUE THREAD

**Issue by Jojoshua** [olifolkerd/tabulator#1883]
In some cases I would want to present an empty table(or table with 1 empty row) which the user will fill out. Instead of creating an explicit button to add a new row, I would like to have an option which automatically adds a new empty row when the user presses the Tab key (only when they are in the last row & cell). This is a type of function that is standard behavior in Excel-like tables.

**Describe alternatives you've considered**
Possibly extending the keybindings module.

**Comment by Jojoshua** [comment]
Hey @olifolkerd , what do you think of this kind of feature?

**Comment by olifolkerd** [comment]
Hey @Jojoshua 

I have implemented this functionality in the past using custom editors to habdle the transition, but i can see it being a usefull feature to add.

I will add it to the roadmap for 4.3

Cheers

Oli :)

**Comment by olifolkerd** [comment]
Hey @Jojoshua 

You will be happy to hear i have just pushed this functionality to the 4.4 branch which should be released in the next couple of weeks.

Cheers

Oli :)

**Comment by Jojoshua** [comment]
Hey @olifolkerd thanks for the enhancement. I was just checking it out and it appears that the new row is added to the table, but the focus is reset to the old row's first cell instead of the new row's first cell. Is this what you had in mind or is it a bug? I was thinking typical excel-like functionality would focus to the new row.

**Comment by olifolkerd** [comment]
Hey @Jojoshua 

It works corretly for me, moving on to the first editable cell on the new row. If you are having trouble with a feature can you please make sure you post a link to a JS fiddle as it is impossible to diagnoe the issue without more info.

Cheers

Oli :)

**Comment by Jojoshua** [comment]
Hey @olifolkerd , here is a fairly simple fiddle to help diagnose. When entering data into the 2nd cell, then tabbing it goes back to the first row cell 1. 

https://jsfiddle.net/jojoshua/Lkvn1pmw/7/

Thanks!

**Comment by olifolkerd** [comment]
Hey @Jojoshua 

Thanks for the example, the issue is because you dont have the virtual DOM enable in the table, and the tabEndNewRow was only really designed to work with the virtualDOM enabled.

I would be happy to look into adding this for classic render mode tables, bu i would expect the table to be used with the Virtual DOM in most cases.

On a side note could you use a bit of indentation on your examples, it makes your code really hard to follow when it is one giant block with out spaces or returns. also there is no need for jQuery in that example.

Cheers

Oli :)

**Comment by Jojoshua** [comment]
Thanks @olifolkerd . In all of my uses for Tabulator I always want all the rows to show at once so I don't want to set a height to enable Virtual DOM. Tabulator's configuration of editors and formatters and the ability to apply the configuration saved from database is much more important to me than rendering thousands of rows. Tabulator has become a new type of form element for capturing data -- much more useful in that regard than displaying already captured data.

You should check with your user base to determine the most common use cases. I hope the classic rendering mode stays well supported.

Sorry about the condensed JSON. It came straight from the database that way, I'll format it next time.

**Comment by Jojoshua** [comment]
Hey @olifolkerd, what is the difference between the enhancement/feature label?

**Comment by olifolkerd** [comment]
Hey @Jojoshua 

Most users use it with the virtual DOM enabled, i spend a lot of time looking at the most common usage cases to meet the needs of Tabulator users, it just seems that you use a lot of the edge features.

An enhancement is an addition to an existing feature or a small upgrade, a feature is a large body of work in its own right that is not enhancing an existing functionality, for example a new module.

It is on the road map to look at adding this for the classic render mode in the next release

Cheers

Oli :)

**Comment by olifolkerd** [comment]
Hey @Jojoshua 

I have pushed a fix for the classic render mode new row creation to the master branch and will include it in a patch release later today.

Cheers

Oli :)
